openwisp_radius/admin.py

```diff
@@ -498,7 +504,22 @@ def has_delete_permission(self, request, obj=None):
         return False
 
 
-UserAdminInline.inlines += [RadiusUserGroupInline, PhoneTokenInline]
+class RegisteredUserInline(StackedInline):
+    model = RegisteredUser
+    form = AlwaysHasChangedForm
+
+    def has_add_permission(self, request, obj):
+        return False
+
+    def has_delete_permission(self, request, obj=None):
+        return False
+
+
+UserAdminInline.inlines += [
+    RadiusUserGroupInline,
+    PhoneTokenInline,
+    RegisteredUserInline,
+]
```

Comment:
I was trying to do some improvements to this file:

```diff
diff --git a/openwisp_radius/admin.py b/openwisp_radius/admin.py
index b5a78de..bcb3c43 100644
--- a/openwisp_radius/admin.py
+++ b/openwisp_radius/admin.py
@@ -9,8 +9,7 @@ from django.forms.widgets import Select
 from django.urls import reverse
 from django.utils.translation import gettext_lazy as _
 
-from openwisp_users.admin import OrganizationAdmin
-from openwisp_users.admin import UserAdmin as UserAdminInline
+from openwisp_users.admin import OrganizationAdmin, UserAdmin
 from openwisp_users.multitenancy import MultitenantAdminMixin, MultitenantOrgFilter
 from openwisp_utils.admin import (
     AlwaysHasChangedMixin,
@@ -507,18 +506,16 @@ class PhoneTokenInline(TimeReadonlyAdminMixin, StackedInline):
 class RegisteredUserInline(StackedInline):
     model = RegisteredUser
     form = AlwaysHasChangedForm
-
-    def has_add_permission(self, request, obj):
-        return False
+    extra = 0
 
     def has_delete_permission(self, request, obj=None):
         return False
 
 
-UserAdminInline.inlines += [
+UserAdmin.inlines.insert(0, RegisteredUserInline)
+UserAdmin.inlines += [
     RadiusUserGroupInline,
     PhoneTokenInline,
-    RegisteredUserInline,
 ]
 
 
@@ -663,7 +660,7 @@ if app_settings.SOCIAL_LOGIN_ENABLED:
         def has_delete_permission(self, request, obj=None):
             return False
 
-    UserAdminInline.inlines += [SocialAccountInline]
+    UserAdmin.inlines += [SocialAccountInline]
 
     Token = apps.get_model('authtoken', 'Token')
```

But one test fails in the SAMPLE_APP:

```
======================================================================
FAIL: test_admin_change_non_superuser_readonly_fields (openwisp2.sample_users.tests.TestUsersAdmin) [User Permissions]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/nemesis/Code/openwisp/openwisp-users/openwisp_users/tests/test_admin.py", line 239, in test_admin_change_non_superuser_readonly_fields
    self.assertContains(response, html)
  File "/home/nemesis/.virtualenvs/openwisp2/lib/python3.8/site-packages/django/test/testcases.py", line 470, in assertContains
    self.assertTrue(real_count != 0, msg_prefix + "Couldn't find %s in response" % text_repr)
AssertionError: False is not true : Couldn't find '<div class="readonly">sample_users' in response

```
Any idea why?

Reply:
Probably the same reason as https://github.com/openwisp/openwisp-users/pull/232, did you update `openwisp-users` in your local repo?